### PR TITLE
fix(smime): add missing primary key to certificate table

### PR DESCRIPTION
**Pull request by st3iny**

### CODE REVIEW

**Comment by st3iny** [comment]
No backports are needed as this is an unreleased feature.

**Review by ChristophWurst — APPROVED** [review]

File: lib/Migration/Version2300Date20230120085320.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+namespace OCA\Mail\Migration;
+
+use Closure;
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version2300Date20230120085320 extends SimpleMigrationStep {
+
```
Comment (ChristophWurst) [inline lib/Migration/Version2300Date20230120085320.php:35]: cs says :fire:

Comment (st3iny) [reply lib/Migration/Version2300Date20230120085320.php]: Done

File: lib/Migration/Version2300Date20230120085320.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+use OCP\Migration\SimpleMigrationStep;
+
+class Version2300Date20230120085320 extends SimpleMigrationStep {
+	/**
+	 * @param IOutput $output
+	 * @param Closure(): ISchemaWrapper $schemaClosure
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
```
Comment (ChristophWurst) [inline lib/Migration/Version2300Date20230120085320.php:41]: > ERROR: MoreSpecificImplementedParamType - lib/Migration/Version2300Date20230120085320.php:41:56 - Argument 2 of OCA\Mail\Migration\Version2300Date20230120085320::changeSchema has the more specific type 'Closure():OCP\DB\ISchemaWrapper', expecting 'Closure' as defined by OCP\Migration\SimpleMigrationStep::changeSchema (see https://psalm.dev/140)

Comment (ChristophWurst) [reply lib/Migration/Version2300Date20230120085320.php:41]: It is not your fault, see https://github.com/nextcloud/server/pull/35618.

File: lib/Migration/Version2300Date20230120085320.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+use Closure;
+use OCP\DB\ISchemaWrapper;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version2300Date20230120085320 extends SimpleMigrationStep {
+	/**
+	 * @param IOutput $output
+	 * @param Closure(): ISchemaWrapper $schemaClosure
```
Comment (ChristophWurst) [inline lib/Migration/Version2300Date20230120085320.php:37]: drop that line and Psalm should pass with stable25

**Comment by st3iny** [comment]
Ok, this PR is cursed.

File: lib/Migration/Version2300Date20230120085320.php (hunk 2 above)
Comment (st3iny) [reply lib/Migration/Version2300Date20230120085320.php:41]: Done

File: lib/Migration/Version2300Date20230120085320.php (hunk 3 above)
Comment (st3iny) [reply lib/Migration/Version2300Date20230120085320.php]: Done

**Pull request merged**